libcaf_core/caf/detail/ring_buffer.hpp

```diff
@@ -0,0 +1,85 @@
+// This file is part of CAF, the C++ Actor Framework. See the file LICENSE in
+// the main distribution directory for license terms and copyright or visit
+// https://github.com/actor-framework/actor-framework/blob/master/LICENSE.
+
+#pragma once
+
+#include "caf/config.hpp"
+
+#include <memory>
+
+namespace caf::detail {
+
+// A simple ring buffer implementation
+template <class T>
+class ring_buffer {
+public:
+  explicit ring_buffer(size_t max_size)
+    : max_size_(max_size), size_(0), write_pos_(0) {
```

Comment:
In visual studio I observed that write_pos_ and size_ were initialized with garbage value which introduces undefined behavior, I chose to keep it in this way for that reason.